docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,85 @@
+# Proposal B
+
+Add "reference" field to existing schemas
+
+## Description
+
+The following proposal describes adding a new field to the existing JSON schemas to enable the creation of relationships between objects stored in an OCI registry.
+
+Additionally it describes a new read-only endpoint on the registry HTTP API to allow listing of all objects which reference a given object.
+
+## Links
+
+| Description                        | Link                                                            |
+| ---------------------------------- | --------------------------------------------------------------- |
+| Original proposal (image-spec)     | [View](https://github.com/opencontainers/image-spec/issues/827) |
+| PR to implement (image-spec)       | [View](https://github.com/opencontainers/image-spec/pull/828)   |
+| Fork of distribution with support  | [View](https://github.com/dlorenc/distribution/tree/references) |
+| Diagram of usage pattern           | [View](https://user-images.githubusercontent.com/1007786/121441279-5c61ca00-c93e-11eb-8b18-40b312542044.png) |
+
+## Modifications
+
+### JSON Schema
+
+All existing supported JSON schemas (index/manifest/descriptor) will be allowed to include a new field, `reference`, which is itself a [descriptor](https://github.com/opencontainers/image-spec/blob/main/specs-go/v1/descriptor.go#L22) object:
+
+```jsonc
+{
+  "mediaType": "icecream/scoops.vanilla.v1.json",
+  "size": 2345,
+  "digest": "sha256:b2b2b2...",
+  "reference": { // <----- New field
+    "mediaType": "icecream/cone.v1.json",
+    "size": 1234,
+    "digest": "sha256:a1a1a1..."
+  }
+}
+```
+
+### Registry HTTP API
+
+A single, read-only endpoint will be added to the registry HTTP API:
+
+```
+GET /v2/<name>/manifests/<ref>/references
+```
+
+The response will be a valid [index](https://github.com/opencontainers/image-spec/blob/main/specs-go/v1/index.go#L21), containing a `manifests` array, which is the complete list of descriptor objects referencing a given `<ref>` (tag or digest) on a given `<name>` (repo).
```

Comment:
It's worth calling out, to contrast with other proposals, that this will return one level of references, not a whole tree.

So if there's an image with a signature, and an SBOM, and that SBOM has a signature, it will require multiple API calls to discover all of those. If you want to be sure you have the whole tree, you have to call `/references` on each leaf to make sure there aren't more leafs under them.

To be clear, I think that's totally fine -- it vastly simplifies the API, and it may even be preferable for clients that don't want to discover the whole tree all the time, or registries that don't want to serve it all the time. It also simplifies server-side limiting of that tree, since it's assumed to be up to the client to only ask for what they care about.